### PR TITLE
feat: Add AutoML vision, Custom training job, and generic prediction samples

### PR DESCRIPTION
**Pull request by andrewferlitsch**
MBSDK samples + tests

### CODE REVIEW

**Comment by vinnysenthil** [comment]
@googlebot I consent.

File: samples/model-builder/custom_training_job_sample.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from google.cloud import aiplatform
+
+#  [START aiplatform_sdk_custom_training_job_sample]
+def custom_training_job_sample(
+    project,
```
Comment (sasha-gitg) [inline samples/model-builder/custom_training_job_sample.py:20]: Should these have type signatures?

File: samples/model-builder/custom_training_job_sample.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    deploy_image,
+    requirements,
+    replica_count,
+):
+    aiplatform.init(project=project, location=location, staging_bucket=bucket)
+
+    job = aiplatform.CustomTrainingJob(
+        display_name=display_name,
+        script_path=script_path,
+        container_uri=train_image,
```
Comment (sasha-gitg) [inline samples/model-builder/custom_training_job_sample.py:36]: Should the parameter names of the sample be consistent with the parameter names in the MB SDK (in which we try to be consistent with the underlying API)? I can see some value in communicating a different name if we feel like the current parameter name is not descriptive enough for a sample.

File: samples/model-builder/custom_training_job_sample.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from google.cloud import aiplatform
+
+#  [START aiplatform_sdk_custom_training_job_sample]
+def custom_training_job_sample(
```
Comment (sasha-gitg) [inline samples/model-builder/custom_training_job_sample.py:19]: Will we (or do we) have a custom_training_pipeline_sample that also takes a managed dataset?

File: samples/model-builder/custom_training_job_sample_test.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    custom_training_job_sample.custom_training_job_sample(
+        project=constants.PROJECT,
+        location=constants.LOCATION,
+        bucket=constants.STAGING_BUCKET,
+        display_name=constants.DISPLAY_NAME,
+        script_path=constants.PYTHON_PACKAGE,
+        script_args=constants.PYTHON_PACKAGE_CMDARGS,
+        train_image=constants.TRAIN_IMAGE,
+        deploy_image=constants.DEPLOY_IMAGE,
+        requirements="",
```
Comment (sasha-gitg) [inline samples/model-builder/custom_training_job_sample_test.py:33]: requirements should be a list

**Review by sasha-gitg — APPROVED** [review]
Added a few comments. Also, I notice some samples use asynchronous work flow while others don't. Is that intended?

File: samples/model-builder/custom_training_job_sample.py (hunk 1 above)
Comment (andrewferlitsch) [reply samples/model-builder/custom_training_job_sample.py]: updated

File: samples/model-builder/custom_training_job_sample.py (hunk 3 above)
Comment (andrewferlitsch) [reply samples/model-builder/custom_training_job_sample.py:19]: not on my assigned sample list

File: samples/model-builder/custom_training_job_sample.py (hunk 2 above)
Comment (andrewferlitsch) [reply samples/model-builder/custom_training_job_sample.py:36]: fixed

File: samples/model-builder/custom_training_job_sample_test.py (hunk 4 above)
Comment (andrewferlitsch) [reply samples/model-builder/custom_training_job_sample_test.py]: fixed

File: samples/model-builder/endpoint_predict_sample.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from google.cloud import aiplatform
+
+#  [START aiplatform_sdk_endpoint_predict_sample]
+def endpoint_predict_sample(project : str, 
+                            location : str, 
+                            instances : str,
```
Comment (dizcology) [inline samples/model-builder/endpoint_predict_sample.py:21]: The type for instances should be a list.  Do we plan to have specialized prediction samples that show how to structure the instances at least for AutoML models?

Comment (andrewferlitsch) [reply samples/model-builder/endpoint_predict_sample.py]: fixed

**Review by vinnysenthil — APPROVED** [review]

**Pull request merged**